### PR TITLE
Change to ConditionalOnProperty, also prefix the cpu count metric

### PR DESCRIPTION
**Pull request by checketts**
Adresses #200

### CODE REVIEW

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public ProcessorMetrics(Iterable<Tag> tags) {
     public void bindTo(MeterRegistry registry) {
         Runtime runtime = Runtime.getRuntime();
 
-        Gauge.builder("cpu", runtime, Runtime::availableProcessors)
+        Gauge.builder("system.cpu.count", runtime, Runtime::availableProcessors)
```
Comment (checketts) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java]: Can we also rename this metric @mweirauch @jkschneider so it uses the hierarchy?

**Comment by checketts** [comment]
@spencergibb I re-added `@ConditionalOnBean` and left the class out, since it is implied by the return type. Is that correct?

**Comment by spencergibb** [comment]
Yes

**Comment by jkschneider** [comment]
Thanks @checketts.

**Pull request merged**